client/python/gradio_client/client.py

```diff
@@ -1171,7 +1171,11 @@ def process_input_files(self, *data) -> tuple:
                     or utils.is_http_url_like(f),
                 )
             elif not self.client.upload_files:
-                d = utils.traverse(d, self._upload_file, utils.is_file_obj_with_meta)
+                d = utils.traverse(
```

Comment:
This was an pre-existing bug where _upload_file was not being passed the index parameter